### ISSUE TITLE
[Bug] Search Torrent doesn't work properly

### ISSUE THREAD

**Issue by burekas7** [elgatito/plugin.video.elementum#524]
## Steps to Reproduce (for bugs)
1) Go to "Search Torrent" -> New Search
2) A keyboard dialog will show up.
3) Type whatever you need, for example in the log: "Vikings S03E01".
4) Confirm
5) The dialog is closed, the text that you typed is added to the History search list, but the search hasn't started.
6) Press on the "Vikings S03E01" and the search process is starting.
7) Pressing on one of the result, the torrent isn't starting.
8) Pressing again on the text from the History Search, Elementum recognize that I have already chose one, I confirm, and nothing happen.

* Kodi Debug Log:
https://paste.ubuntu.com/p/nRrmzscSjj

## Expected Behavior
1. The search process should be start right after confirm the text in the keyboard dialog. 
2. The torrent download should start.

## Your Environment
* Elementum: 0.1.43
* Environment name and version: Kodi 17.6
* Operating System and version: Windows 10, 64bit

**Comment by elgatito** [comment]
I cannot reproduce your issue on default skin.

**Comment by burekas7** [comment]
I reproduce it also with the default skin: Estuary

On this case I reproduce steps 1-5
And also steps 6-7
Sometimes it do success step 5 and burst start searching but sometimes it's just added to the history search list without do anything.
I don't know why, maybe the reason that it remember last result selection relevant, I don't know.

Here is the debug log:
https://paste.ubuntu.com/p/mr2RJMDH6v

**Comment by elgatito** [comment]
@burekas7 
Should be fixed in 0.1.48.

**Comment by burekas7** [comment]
Thanks but still the same with 0.1.48 as my last comments here:

When steps 5 isn't working:
Kodi Log (Debug) - https://paste.ubuntu.com/p/s5x6fPybrw

When steps 1-5 are ok (Except that there is a delay since the searching text confirmed and the Burst actually starting the searching)
And the problem is still as described in steps 6-7.
Kodi Log: https://paste.ubuntu.com/p/FDqqCZm4fT

---
Elementum: 0.1.48
Elementum Burst: 0.0.46
Environment name and version: Kodi 17.6
Skin: Titan
Operating System and version: Windows 10, 64bit

**Comment by elgatito** [comment]
I don't have any problems with starting search after it is entered by keyboard.
Anyway, I was just doing few hot-fixes for file storage and updated versions.

**Comment by elgatito** [comment]
@burekas7 you can also try 0.1.49, but I don't know what happens on your side.

**Comment by burekas7** [comment]
The same with 0.1.49 :\

Kodi Debug log:
https://paste.ubuntu.com/p/JJSTKdkG8k

**Comment by antonsoroko** [comment]
@burekas7 
looks like you ran into some very rare bug.
one option would be to remove elementum (not kodi) database (`storm.db` iirc) file and basically start from the scratch. 
or if you feel fancy you can compile https://github.com/br0xen/boltbrowser and use it to open storm.db and remove everything except setting and torrents info - this might help (or not :-) ).

or maybe you have set some UI settings (like timeouts) that uncover some rare bug.

just thinking out loud.

**Comment by adamcarter81** [comment]
I am also suffering this issue - default skin, no customisations, kodi 18.6 on RPI3, Libre Elec build 9.2.1
If I search by entering text, nothing happens but the search request does show in the search history most of the time, and then selecting that it searches normally.

The only log entry related I can see is:

2020-04-23 07:32:12.108 T:1740579712   ERROR: GetDirectory - Error getting plugin://plugin.video.elementum/search?keyboard=1
2020-04-23 07:32:12.111 T:1937399824   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.elementum/search?keyboard=1) failed

**Comment by adamcarter81** [comment]
Update - solved by clearing the database under Advanced settings

**Comment by burekas7** [comment]
@adamcarter81 
I have tried this, but there is no change.
It's probably another issue that I need to test.
Sometimes it works so there is something unstable.
Sometimes pressing the 'escape' key right after steps 4 or 7 make it works but also quit from the addon menu, I don't know why.

**Comment by burekas7** [comment]
Here some videos to demonstrate the problem and the 'escape' button which helps for unknown reason:
* When everything is ok (Sometimes it happens :) )
https://f2h.io/m9xh71v104xw
* When the search is responding ok but when trying to play, nothing happens.
https://f2h.io/v77xzqtqj5qd
* When the search is responding ok and pressing the 'escape' key right after choosing one of the result helps and make it works (but also make the addon quit from the menu)
https://f2h.io/d4b77xdzghe3

I'm trying to figure out why and will make some test in different conditions and also with clean kodi.

**Comment by elgatito** [comment]
@burekas7 Please, try with the latest version and post a new log.

**Comment by burekas7** [comment]
@elgatito 

So I tried and tested it with 0.1.52.
I run these searches:
Money Heist S01E01
Money Heist S01E02
Money Heist S01E03
Money Heist S01E04
Money Heist S01E05
Money Heist S01E06
Money Heist S01E07
Money Heist S01E08

For episodes: 5,8 , the search text just added to the Search History list and not start the search process automatically after typing the text and confirm.
For episodes: 1,4,7 , the search start right after typing the search text but didn't played the torrnt after choosing it from the results.
All the rest works as should be, search starts right after typing and confirm, and selecting the torrent from the results start the torrent download and play.
So there is stiil a little instability, so currently if there is a case that it's not working well, I just make a new search until it's working.

Kodi debug log:
https://paste.ubuntu.com/p/Sd22hmP3m7

From tests I made with a clean kodi build, I start to think that it's a problem that related to the skin (Official Titan for 17.6, 3.7.x).
But to be sure I need to do a lot of tests, so maybe there is also another issue.
